### ISSUE TITLE
Conditional is checked even if corresponding node is not running

### ISSUE THREAD

**Issue by ijisthee** [acdamiani/schema#18]
Hi @acdamiani 

how to reproduce?
![image](https://github.com/acdamiani/schema/assets/20422129/001f8bf2-1006-4fcb-bd87-676ed22134a4)

In this example I would **expect** that **only two checks** are made since the tree should never reach
![image](https://github.com/acdamiani/schema/assets/20422129/1f1718fe-844c-4c74-ba64-5c70b4eae7fa)

But unfortunately all three conditionals are checked.
Only when a conditional has abort conditions.

If this is intended I would like to understand the advantage.

Best,
Christian

**Comment by acdamiani** [comment]
Yep, this is an oversight. Try a `git checkout` on this commit: eccf515abd6a0b430046cfbbe9cf9e284935fe5d. Let me know if your issue is fixed; I want to do some more testing before merging the associated pull request.

**Comment by ijisthee** [comment]
Cannot really test it, since this issue is back again: https://github.com/acdamiani/schema/issues/5

**Comment by acdamiani** [comment]
I just pushed a fix for the resurgence of #5. I'll do some more testing for #19 to make sure this issue is fixed before merging it.
